monkey/monkey_island/cc/resources/events.py

```diff
@@ -11,10 +14,26 @@
 class Events(AbstractResource):
     urls = ["/api/events"]
 
+    def __init__(
+        self,
+        agent_event_queue: IAgentEventQueue,
+        event_serializer_registry: EventSerializerRegistry,
+    ):
+        self._agent_event_queue = agent_event_queue
+        self._event_serializer_registry = event_serializer_registry
+
     # Agents needs this
     def post(self):
-        event = request.json
+        events = request.json
+
+        for event in events:
+            try:
+                serializer = self._event_serializer_registry[event["type"]]
+                deserialized_event = serializer.deserialize(event)
+            except (TypeError, ValidationError) as err:
+                logger.debug(f"Error occured deserializing an event {event}: {err}")
+                return {"error": str(err)}, HTTPStatus.INTERNAL_SERVER_ERROR
```

Comment:
5xx errors mean the server had an issue, 4xx errors mean the user did something wrong (like send unparsable data). The reponse code should be https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400.